onnx/test/runtime_test.py

```diff
@@ -0,0 +1,1174 @@
+            out = stdout.getvalue()
+            log = dedent(
+                """
+                Add(B01, B11) -> C0
+                -- begin Add.run(2 inputs)
+                -- done Add.run -> 1 outputs
+                 + C0: float32:(2, 2):[4.0, 6.0, 8.0, 10.0]
+                Sub(B11, B21) -> C1
+                -- begin Sub.run(2 inputs)
+                -- done Sub.run -> 1 outputs
+                 + C1: float32:(2, 2):[8.0, 10.0, 12.0, 14.0]
+                Mul(C0, C1) -> D0
+                -- begin Mul.run(2 inputs)
+                -- done Mul.run -> 1 outputs
+                 + D0: float32:(2, 2):[32.0, 60.0, 96.0, 140.0]
+                """
+            ).lstrip("\n")
+            self.assertEqual(log, out)
+
+    def test_inference_lr(self):
+        lr, f = TestRuntimeInference._linear_regression()
+        x = np.array([[0, 1], [2, 3]], dtype=np.float32)
+        a = np.array([1, 1], dtype=np.float32)
+        b = np.array([11], dtype=np.float32)
+        expected = f(x, a, b)
+        sess = rt.Inference(lr)
+        got = sess.run(None, {"X": a, "A": a, "B": b})[0]
+        assert_almost_equal(expected, got)
+
+    def test_inference_lr_clip(self):
+        with self.subTest(opt="min+max"):
+            lr, f = TestRuntimeInference._linear_regression(clip=True)
+            x = np.array([[0, 1], [2, 3]], dtype=np.float32)
+            a = np.array([1, 1], dtype=np.float32)
+            b = np.array([11], dtype=np.float32)
+            expected = f(x, a, b)
+            sess = rt.Inference(lr)
+            last_node = sess.rt_nodes_[-1]
+            self.assertEqual(last_node.__class__.__name__, "Clip_11")
+            got = sess.run(None, {"X": a, "A": a, "B": b})[0]
+            assert_almost_equal(expected, got)
+
+        with self.subTest(opt="max"):
+            lr, f = TestRuntimeInference._linear_regression(clip=True, min_value=None)
+            x = np.array([[0, 1], [2, 3]], dtype=np.float32)
+            a = np.array([1, 1], dtype=np.float32)
+            b = np.array([11], dtype=np.float32)
+            expected = f(x, a, b)
+            sess = rt.Inference(lr)
+            last_node = sess.rt_nodes_[-1]
+            self.assertEqual(last_node.__class__.__name__, "Clip_11")
+            got = sess.run(None, {"X": a, "A": a, "B": b})[0]
+            assert_almost_equal(expected, got)
+
+        with self.subTest(opt="min"):
+            lr, f = TestRuntimeInference._linear_regression(clip=True, max_value=None)
+            x = np.array([[0, 1], [2, 3]], dtype=np.float32)
+            a = np.array([1, 1], dtype=np.float32)
+            b = np.array([11], dtype=np.float32)
+            expected = f(x, a, b)
+            sess = rt.Inference(lr)
+            last_node = sess.rt_nodes_[-1]
+            self.assertEqual(last_node.__class__.__name__, "Clip_11")
+            got = sess.run(None, {"X": a, "A": a, "B": b})[0]
+            assert_almost_equal(expected, got)
+
+    def test_inference_lr_clip_6(self):
```

Comment:
Doesn't have to be this PR - Looks like much of these tests can be parameterized: https://pypi.org/project/parameterized/

Reply:
parametized is not a dependency. I suggest another PR.